src/idd/idd.markers.uncertain.js

```diff
@@ -0,0 +1,1279 @@
+                border = "#000000";
+                color = "#ffffff";
+                drawBorder = true;
+            }
+            else {
+                color = data.color;
+                border = color;
+                if (data.border != null) {
+                    drawBorder = true;
+                    border = data.border;
+                }
+            }
+        };
+
+        var renderShape = function () {
+            var sampleColor = "gray";
+            var sampleBorderColor = "gray";
+
+            InteractiveDataDisplay.Petal.drawSample(context, x1, y1, halfSize / 2, halfSize, sampleColor);
+        };
+
+        refreshColor();
+        refreshSize();
+        renderShape();
+    }
+};
+InteractiveDataDisplay.BullEye = {
+    prepare: function (data) {
+        data.bullEyeShape = data.bullEyeShape ? data.bullEyeShape.toLowerCase() : "circle";
+        // y
+        if (data.y == undefined || data.y == null) throw "The mandatory property 'y' is undefined or null";
+        if (!InteractiveDataDisplay.Utils.isArray(data.y)) throw "The property 'y' must be an array of numbers";
+        var n = data.y.length;
+
+        var mask = new Int8Array(n);
+        InteractiveDataDisplay.Utils.maskNaN(mask, data.y);
+
+        // x
+        if (data.x == undefined)
+            data.x = InteractiveDataDisplay.Utils.range(0, n - 1);
+        else if (!InteractiveDataDisplay.Utils.isArray(data.x)) throw "The property 'x' must be an array of numbers";
+        else if (data.x.length != n) throw "Length of the array which is a value of the property 'x' differs from lenght of 'y'"
+        else InteractiveDataDisplay.Utils.maskNaN(mask, data.x);
+
+        // border
+        if (data.border == undefined || data.border == "none")
+            data.border = null; // no border
+
+        // colors        
+        if (data.color == undefined) data.color = InteractiveDataDisplay.Markers.defaults.color;
+        if (InteractiveDataDisplay.Utils.isArray(data.l95) && InteractiveDataDisplay.Utils.isArray(data.u95)) {
+            if (data.l95.length != n && data.u95.length != n) throw "Length of the array 'color' is different than length of the array 'y'"
+            if (n > 0 && typeof (data.l95[0]) === "number" && typeof (data.u95[0]) === "number") { // color is a data series                 
+                var palette = data.colorPalette;
+                if (palette == undefined) palette = InteractiveDataDisplay.Markers.defaults.colorPalette;
+                if (typeof palette == 'string') palette = new InteractiveDataDisplay.ColorPalette.parse(palette);
+                if (palette != undefined && palette.isNormalized) {
+                    var r = { min: InteractiveDataDisplay.Utils.getMin(data.l95), max: InteractiveDataDisplay.Utils.getMax(data.u95) };
+                    r = InteractiveDataDisplay.Utils.makeNonEqual(r);
+                    data.colorPalette = palette = palette.absolute(r.min, r.max);
```

Comment:
@ElenaPochernina Make u95 and l95 containing color strings, not numbers.
Remove building colors from `draw`